version: "3.8"


services:


  remote-code-db:
    container_name: remote-code-db
    image: postgres

    env_file:
      - service_db/runtime.env

    environment:
      PGDATA: '/var/lib/postgresql/data/pgdata'

    expose:
      - "5432"

    networks:
      remote_code_net:

    labels:
      - "traefik.enable=false"


    volumes:
      - type: bind
        source: ./service_db/db_files
        target: /var/lib/postgresql/data/pgdata

      - type: bind
        source: ./service_db/init_script/
        target: /docker-entrypoint-initdb.d

    restart: unless-stopped

  remote-code-master:
    container_name: remote-code-master
    image: remote-code-master:latest
    build:
      context: ./../
      dockerfile: ./core/Dockerfile

    labels:
      - "traefik.http.routers.app.rule=Host(`api.remote-code.woldseth.xyz`) && PathPrefix(`/app/`)"
      - "traefik.docker.network=remote_code_net"

    expose:
      - "9080"
      - "9443"

    env_file:
      - runtime_envs/db_login.env
      - runtime_envs/kafka.env

    volumes:
      - type: bind
        source: $PWD/../../../NFS_SHARED_FILE
        target: /application_storage/shared_files


    healthcheck:
      test: [ "CMD", "curl", "-f", "http://localhost/health/ready" ]
      interval: 1m30s
      timeout: 10s
      retries: 3
      start_period: 10s

    restart: unless-stopped

    networks:
      - remote_code_net

    depends_on:
      - remote-code-db

networks:
  remote_code_net:
    external: true

